napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,520 @@
+"""Triangulation utilities"""
+
+from __future__ import annotations
+
+import numpy as np
+from numba import njit
+
+
+@njit(cache=True, inline='always')
+def _calc_output_size(
+    normals: np.ndarray, closed: bool, cos_miter_limit: float, bevel: bool
+) -> int:
+    """Calculate the size of the output arrays for the triangulation.
+
+    Parameters
+    ----------
+    normals : np.ndarray
+        Nx2 array of normal vectors of the path
+    closed : bool
+        True if shape edge is a closed path.
+    cos_miter_limit : float
+        Miter limit which determines when to switch from a miter join to a
+        bevel join
+    bevel : bool
+        If True, a bevel join is always to be used.
+        If False, a bevel join will only be used when the miter limit is exceeded.
+
+    Returns
+    -------
+    int
+        number of points in the output arrays
+
+    Notes
+    -----
+    We use a miter limit instead of maximum miter length
+    for performance reasons.
+    This approach provides an equivalent check.
+    See note in generate_2D_edge_meshes
```

Comment:
@Czaki sorry that I waited so long after our walkthrough to work on this. I am back at it. 😅 and hope we can get this in ~tomorrow.

In this docstring, I would appreciate definitions and/or links to understand the difference between "miter limit" and "maximum miter length". They sound like exactly the same thing?

Also, "an equivalent check" — what does this mean?

Reply:
Ah. I do not update the name here. 
The `maximum miter length` is the maximum length of miter vector.

The `miter limit` is now `cos_miter_limit` and is the limit for the cosines value. 

It is calculated to be equivalent to checking of vector length, but cheaper to calculate.